ts/kpt-functions/src/types.ts

```diff
@@ -182,12 +184,14 @@ export class Configs {
         `functionConfig expected to be of kind ConfigMap, instead got: ${cm.kind}`
       );
     }
+
     const configMap = new Map<string, string>();
     for (const key in cm.data) {
       if (cm.data.hasOwnProperty(key)) {
         configMap.set(key, cm.data[key]);
       }
     }
+
     return configMap;
```

Comment:
Should we also validate the ConfigMap in `getFunctionConfigMap`?
We can probably throw the same type of error `FunctionConfigError`